diff --git a/kernel/sched/fair.c b/kernel/sched/fair.c
index 0ca1669563cb..2816c821d8b3 100644
--- a/kernel/sched/fair.c
+++ b/kernel/sched/fair.c
@@ -8054,6 +8054,7 @@ static int
 wakeup_preempt_entity(struct sched_entity *curr, struct sched_entity *se)
 {
 	s64 gran, vdiff = curr->vruntime - se->vruntime;
+	s64 offset = wakeup_latency_gran(curr, se);
 #ifdef CONFIG_SCHED_BORE
 	if (likely(sched_bore)) {
 		u64 rtime = curr->sum_exec_runtime - curr->prev_sum_exec_runtime;
@@ -8062,10 +8063,10 @@ wakeup_preempt_entity(struct sched_entity *curr, struct sched_entity *se)
 	}
 #endif // CONFIG_SCHED_BORE
 
-	if (vdiff <= 0)
+	if (vdiff < offset)
 		return -1;
 
-	gran = wakeup_gran(se);
+	gran = offset + wakeup_gran(se);
 
 	/*
 	 * At wake up, the vruntime of a task is capped to not be older than
